kernels/min.cu

```diff
@@ -0,0 +1,145 @@
+#include <cusp/helper_cuda.h>
+#include <complex>
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include <cusp/min.cuh>
+
+#define default_min_block 256
+#define default_min_grid 32
+
+namespace cusp {
+
+template <typename T>
+__global__ void kernel_min(const T* ins, T* out, int stream_number, int grid_size, int N)
+{
+    __shared__ T cache[default_min_block];
+
+    int i = blockDim.x * blockIdx.x + threadIdx.x;
+    int cacheIndex = threadIdx.x;
+
+    T temp = ins[i];
+    while (i < N) {
+    	if(ins[i] < temp)
+    		temp = ins[i];
+        i += blockDim.x * gridDim.x;  
+    }
+   
+    cache[cacheIndex] = temp;
+
+    __syncthreads();
+
+    int ib = blockDim.x / 2;
+    while (ib != 0) {
+      if(cacheIndex < ib && cache[cacheIndex + ib] < cache[cacheIndex])
+        cache[cacheIndex] = cache[cacheIndex + ib]; 
+
+      __syncthreads();
+
+      ib /= 2;
+    }
+    
+    if(cacheIndex == 0) {
+        // out[blockIdx.x] = cache[0];
+        out[blockIdx.x + stream_number * grid_size] = cache[0];
+    }
+}
+
+
+template <typename T>
+__global__ void kernel_get_min_single(T * out, int grid_size, int ninputs) {
+    T max = out[0];
```

Comment:
probably should rename max to min here and other places

Reply:
Yeah that's my bad, I just copied a whole bunch of the code from max and forgot to change the variable names - I'll get on that. 

On another note, I've been recording some benchmarks on thrust versus plain cuda performance for kernels (because many kernels can be written very easily in thrust) and found that for reducible kernels, thrust was consistently 4 or 5 times slower - just goes to show that we were correct in opting to not use thrust for our kernels.